sklearn/ensemble/_hist_gradient_boosting/common.pxd

```diff
@@ -7,6 +7,8 @@ ctypedef np.npy_float64 X_DTYPE_C
 ctypedef np.npy_uint8 X_BINNED_DTYPE_C
 ctypedef np.npy_float64 Y_DTYPE_C
 ctypedef np.npy_float32 G_H_DTYPE_C
+ctypedef np.npy_uint32 X_BITSET_INNER_DTYPE_C
+ctypedef X_BITSET_INNER_DTYPE_C[8] X_BITSET_DTYPE_C
```

Comment:
Also, no need to prepend `X_` to the dtype name since this doesn't directly relate to the input `X`.

Reply:
I wanted to do this, but it did not work because of a circular dependency.

`common.pyd` defines `X_BINNED_DTYPE_C` which is needed by `bitset.pxd` and `common.pyd` needs `BITSET_DTYPE_C` to define `node_struct`.